system/jlib/jmetrics.hpp

```diff
@@ -154,6 +154,11 @@ class jlib_decl CounterMetric : public MetricVal
     {
         value.fetch_add(val);
     }
+
+    void fastInc(uint16_t val = 1)
+    {
+        value = value + val;
```

Comment:
Isn't this actually slower ..?
Because it's still an atomic "value + val" will require 'value' to be read atomically, and then the assignment requires an atomic assignment .
So potentially twice as slow..

@ghalliday - should the fast version instead use memory_order_relaxed order? (but would it actually be any quicker, given 2 ops).

And/or for places where you know the counter is only every going to be single threaded,
should there be a different implementation, e.g. CounterNonAtomicMetric

Reply:
@jakesmith this is actually faster on intel.   The reason is that read and write are actually always atomic (unless they are misaligned, I think over a cache line), and it has a naturally strong memory ordering.  An atomic increment uses a LOCK ADD - which has various effects - mainly synchronizing the cache line over different cores, and has an overhead of ~18cycles as a result (read https://www.agner.org/optimize/instruction_tables.pdf if you really care!).

On non intel chips with a weak memory model there is likely to be a difference between using memory_order_relaxed - which is what the RelaxedAtomic defaults to - because the default std::atomic mode will probably require extra work to ensure the results are consistent (which isn't needed here).

So yes, this is faster, but even better would be to use RelaxedAtomic, or use the memory_order_relaxed on the access of the metric values,

Reply:
ok, it's less than obvious from a casual glance at the code.

Reply:
should now call value.fastAdd(val);

Reply:
Updated to use relaxed atomic class methods.